scalameta/tokenizers/shared/src/main/scala/scala/meta/internal/tokenizers/ScalametaTokenizer.scala

```diff
@@ -60,11 +60,13 @@ class ScalametaTokenizer(input: Input, dialect: Dialect)(implicit options: Token
           pushPart(beg)
           pushToken(Token.Interpolation.SpliceStart(input, dialect, dollarOffset, dollarOffset + 1))
           val splice = nextToken()
-          pushToken(getToken(splice))
-          if (splice.token == LBRACE) loop(braceBalance = 1)
-          val end = nextToken()
-          pushToken(Token.Interpolation.SpliceEnd(input, dialect, end.offset, end.offset))
-          emitContents(end)
+          if (splice.token != STRINGPART) {
+            pushToken(getToken(splice))
+            if (splice.token == LBRACE) loop(braceBalance = 1)
+            val end = nextToken()
+            pushToken(Token.Interpolation.SpliceEnd(input, dialect, end.offset, end.offset))
+            emitContents(end)
+          } else emitContents(splice)
```

Comment:
should we do something like this, perhaps:
```
val end = 
  if (splice.token == STRINGPART) splice
  else {
    pushToken(getToken(splice))
    if (splice.token == LBRACE) loop(braceBalance = 1)
    nextToken()
  }
  pushToken(Token.Interpolation.SpliceEnd(input, dialect, end.offset, end.offset))
  emitContents(end)
```
that way, we always close SpliceStart with a SpliceEnd.

Reply:
Done!